hdfs-executor/src/main/java/org/apache/mesos/hdfs/executor/AbstractNodeExecutor.java

```diff
@@ -127,13 +111,15 @@ private void createSymbolicLink() {
           log.info("Unable to unlink old sym link. Link may not exist. Exit code: " + exitCode);
         }
       } catch (IOException e) {
-        log.fatal("Could not unlink " + hdfsBinaryPath + ": " + e);
-        System.exit(1);
+        log.fatal("Could not unlink " + hdfsBinaryPath, e);
+        throw e;
```

Comment:
You have to follow the logic all the way through...  Part of this work is to reduce the complexity metric.  The number of returns and system.exits increases that metric.   The throw of e here on line 115 is catch by the surround try/catch.   The catch in line 132 has a system exit on line 134.  I usually would be against needing to follow exceptions for logic, however in this case it is within 20 lines of each other and it seems obvious.

let me know if you want to see something different, but there is no bug or serious error here.